chains/txmgr/txmgr.go

```diff
@@ -743,23 +745,52 @@ func (b *Txm[CID, HEAD, ADDR, THASH, BHASH, R, SEQ, FEE]) GetTransactionFee(ctx
 		return fee, fmt.Errorf("failed to find receipt with IdempotencyKey %s", transactionID)
 	}
 
+	txFee := b.CalculateFee(FeeParts{
+		GasUsed:           receipt.GetFeeUsed(),
+		EffectiveGasPrice: receipt.GetEffectiveGasPrice(),
+		L1Fee:             receipt.GetL1Fee(),
+	})
+
+	fee = &evmtypes.TransactionFee{
+		TransactionFee: txFee,
+	}
+
+	return fee, nil
+}
+
+func (b *Txm[CID, HEAD, ADDR, THASH, BHASH, R, SEQ, FEE]) GetTransactionReceipt(ctx context.Context, transactionID string) (receipt *txmgrtypes.ChainReceipt[THASH, BHASH], err error) {
+	foundReceipt, err := b.txStore.FindReceiptWithIdempotencyKey(ctx, transactionID, b.chainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find receipt with IdempotencyKey %s: %w", transactionID, err)
+	}
+
+	// This check is required since a no-rows error returns nil err
+	if foundReceipt == nil {
+		return nil, fmt.Errorf("failed to find receipt with IdempotencyKey %s", transactionID)
+	}
+	return &foundReceipt, nil
+}
```

Comment:
I don't think that this needs a ptr?
```suggestion
func (b *Txm[CID, HEAD, ADDR, THASH, BHASH, R, SEQ, FEE]) GetTransactionReceipt(ctx context.Context, transactionID string) (receipt txmgrtypes.ChainReceipt[THASH, BHASH], err error) {
	foundReceipt, err := b.txStore.FindReceiptWithIdempotencyKey(ctx, transactionID, b.chainID)
	if err != nil {
		return nil, fmt.Errorf("failed to find receipt with IdempotencyKey %s: %w", transactionID, err)
	}

	// This check is required since a no-rows error returns nil err
	if foundReceipt == nil {
		return nil, fmt.Errorf("failed to find receipt with IdempotencyKey %s", transactionID)
	}
	return foundReceipt, nil
}
```